### ISSUE TITLE
Multi Monitoring

### ISSUE THREAD

**Issue by Setsuneh** [firehol/netdata#868]
Hi,

Any version for NETDATA in multi Monitoring?

Thanks.

**Comment by LucaDev** [comment]
Take a look at custom Dashboards - especially [tv.html](https://github.com/firehol/netdata/blob/master/web/tv.html) for now. But there are plans for a central netdata instance

**Comment by Setsuneh** [comment]
Yep thanks.
But you must install on each dedicated servers Netdata, i don't like that sorry :)

**Comment by ktsaou** [comment]
> But you must install on each dedicated servers Netdata, i don't like that sorry :)

Dear @Setsuneh, this is a meaningless statement. All monitoring solutions install an agent on each host to collect data. There is no agentless monitoring. You are missing the point. 

Netdata is an attempt to avoid the central monitoring infrastructure by embedding all the functions on the agent that is installed on all servers, exploring the great features that are possible due to this design (e.g. real-time performance monitoring, thousands of metrics collected, etc) and requiring minimal instances for the central components (check for example the registry).

All the features you probably like in netdata, come from this design. Light, distributed monitoring, embedded in the servers, instead of centralized resource hungry, dedicated monitoring infrastructure.

Netdata is very fast and light. Probably faster and lighter than the agents required by most open source and commercial solutions.

If however you don't like our design, there are many great traditional monitoring solutions, open source and commercial, you can use.

**Comment by anurag03** [comment]
@ktsaou I am not an expert but I feel Nagios does that?

**Comment by ktsaou** [comment]
> @ktsaou I am not an expert but I feel Nagios does that?

No system does that. It is just not possible. The Linux kernel does not expose this information over the net.

A few solutions (like [rtop](http://www.rtop-monitor.org/), [ossec](https://ossec.github.io/docs/manual/agent/agentless-monitoring.html), etc), instead of installing an agent, they `ssh` to the servers to collect the values they need. I think this is totally unacceptable (and they actually turn `sshd` to a monitoring agent).

Other solutions ([including nagios](https://www.nagios.com/solutions/agentless-monitoring/)) call *agentless monitoring* their ability to collect metrics over a third party agent (SNMP for Linux, WMI for Windows).

Even for WMI, the fact that the agent is shipped as standard by the O/S, does not mean there is no agent. WMI is a service (the corresponding of a Linux daemon). So, even in Windows, the agent is there.

So, an agent is always required. The fact that the agent is third party, does not mean there is no agent at all.

A few cloud providers also use the term *agentless monitoring* to emphasize the fact that they can provide a (very limited) set of metrics via their hypervisors. netdata does this too for containers and VMs, if you install netdata at the host (so you can monitor each container and VM from the host, without installing anything at the container or VM).

In short, there is no agentless monitoring. An agent is always required, though it may be shipped by another party.

netdata is a monitoring agent. The netdata API can be used by other solutions to fetch the data they need (for example, nagios could support collecting data via netdata, and they will call it **agentless**, since they don't provide the agent).

**Comment by Ferroin** [comment]
'Agentless' as used not just by Nagios and other monitoring tools, but also orchestration tools like Ansible is not intended to mean that you need nothing on the target system, but that you don't have to install anything dedicated to that purpose on the target system since they leverage existing infrastructure.  Beyond that, it's just a matter of scoping.

Primary examples include:
* Monitoring that a network interface on a given system is online by sending periodic ARP requests for the IP address it's supposed to have is agentless from the perspective of the system being monitored (you're directly leveraging the existing network infrastructure to monitor the system).
* Getting system status of a Windows system via WMI is functionally agentless from the perspective of the target system because WMI is a core component that many other things in the system depend on (and thus you're again leveraging existing infrastructure that's already in-place).
* Monitoring CPU and memory usage of a virtual machine from the host system is agentless from the perspective of the guest system (this is what cloud providers are talking about).

**Comment by ktsaou** [comment]
Yeap. Still the information netdata collects cannot be collected without installing netdata on the target system. netdata is what WMI is for windows, SNMP is for routers, etc.
